mongodump/mongodump_test.go

```diff
@@ -1415,3 +1490,249 @@ func TestCount(t *testing.T) {
 		})
 	})
 }
+
+func TestTimeseriesCollections(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+
+	session, err := testutil.GetBareSession()
+	if err != nil {
+		t.Errorf("could not get session: %v", err)
+	}
+	fcv := testutil.GetFCV(session)
+	if cmp, err := testutil.CompareFCV(fcv, "5.0"); err != nil || cmp < 0 {
+		t.Skip("Requires server with FCV 5.0 or later")
+	}
+
+	colName := "timeseriesColl"
+	dbName := "timeseries_test_DB"
+	err = setUpTimeseries(dbName, colName)
+	if err != nil {
+		t.Errorf("could not setup timeseries collection: %v", err)
+	}
+
+	Convey("With a MongoDump instance", t, func() {
+
+		md := simpleMongoDumpInstance()
+		md.ToolOptions.Namespace.DB = dbName
+		md.OutputOptions.Out = "dump"
+
+		Convey("a timeseries collection should produce a well-formatted dump", func() {
+			Convey("to an archive", func() {
+
+				md.OutputOptions.Out = ""
+				md.OutputOptions.Archive = "dump.archive"
+
+				Convey("when dumping the whole database", func() {
+					// This is the default test of "to an archive".
+					// It's meant to be a noop, do not delete.
+				})
+
+				Convey("when the collection is specified in --collection", func() {
+					md.ToolOptions.DB = dbName
+					md.ToolOptions.Collection = colName
+				})
+
+				Convey("even when the system.buckets collection is excluded", func() {
+					Convey("by --excludeCollections", func() {
+						md.OutputOptions.ExcludedCollections = []string{"system.buckets." + colName}
+					})
+
+					Convey("by --excludeCollectionsWithPrefix", func() {
```

Comment:
Whoa, I didn't know `goconvey` allowed this! I know you could nest `Convey` calls to have a bunch of setup code that is rerun for each inner scope, but you can also have inner `Convey` calls do multiple branches of setup for assertions in an outer scope?

Reply:
Yea, surprising to me at first as well! Seems it reruns the outer scope code for each inner Convey calls!